### PR TITLE
Test improvements, fix testCreateThenInstall flake

### PR DESCRIPTION
**Pull request by martinpitt**
I'm trying to debug https://artifacts.dev.testing-farm.io/19531be9-b707-4fec-8b9b-b4795df9fe0b/ , but the test fails in a mostly useless way. Supposedly the domain XML does not have any metadata, but it could also have a different count. Let's make this more useful in the future.

This also fixes this different [testCreateThenInstall failure](https://artifacts.dev.testing-farm.io/237d681f-3da2-4c04-85d6-d95f597efa93/).

### CODE REVIEW

**Comment by martinpitt** [comment]
I landed the [targetcli naughty](https://github.com/cockpit-project/bots/pull/3164), and c8s should be okay again, so let's re-run packit.

**Comment by martinpitt** [comment]
Oh right, needs rebase.

**Comment by martinpitt** [comment]
Something [went wery wrong](https://artifacts.dev.testing-farm.io/547c0f10-57fc-4d08-a43f-66370c598acf/) with the packit c9s test. No artiffacts. The others were green (so at least rawhide works again :tada: ). I'll wait for @skobyda 's review, if he has change suggestions, they'll re-run anyway.

File: test/check-machines-create
Comment (skobyda) [inline test/check-machines-create:1349]: What is the reason for this sleep?

Comment (martinpitt) [reply test/check-machines-create:1349]: So that it doesn't retry the virsh dumpxml and assertions 5 times in a row without delay -- the purpose is to give it 10 seconds (5 retries times 2s sleep) for the metadata to appear. Unfortunately this is asynchronous -- I first tried to immediately assert that the XML is as expected, but that failed. So the original wrapping into `wait()` was justified.

**Review by skobyda — APPROVED** [review]
Thanks, let's give packit tests one more try?

**Comment by martinpitt** [comment]
@skobyda : c9s failed the same way in #642, so I'm afraid this broke independently. I'm happy to investigate after this and #642 land.

**Pull request merged**
